backend/database/store/redis.go

```diff
@@ -0,0 +1,85 @@
+package store
+
+import (
+	"bytes"
+	"fmt"
+	"log/slog"
+	"os/exec"
+	"strings"
+
+	"github.com/GenerateNU/sac/backend/config"
+)
+
+type Stores struct {
+	Limiter     LimiterInterface
+	Blacklist   BlacklistInterface
+	ActiveToken ActiveTokenInterface
+}
+
+func NewStores(limiter LimiterInterface, blacklist BlacklistInterface, activeToken ActiveTokenInterface) *Stores {
+	return &Stores{
+		Limiter:     limiter,
+		Blacklist:   blacklist,
+		ActiveToken: activeToken,
+	}
+}
+
+func ConfigureRedis(settings config.Settings) *Stores {
+	var activeTokens *ActiveToken
+	var blacklist *Blacklist
+	var limiter *Limiter
+
+	for _, service := range settings.Redis {
+		client := NewRedisClient(service.Username, service.Host, service.Port, service.Password, service.DB)
+		switch service.Username {
+		case "redis_active_tokens":
+			activeTokens = NewActiveToken(client)
+		case "redis_blacklist":
+			blacklist = NewBlacklist(client)
+		case "redis_limiter":
+			limiter = NewLimiter(client)
+		default:
+			slog.Error(fmt.Sprintf("unknown redis service: %s", service.Username))
+		}
+	}
+
+	stores := NewStores(limiter, blacklist, activeTokens)
+
+	EstablishConn(settings)
+
+	return stores
+}
+
+func EstablishConn(settings config.Settings) {
+	services := settings.Redis
+
+	for _, service := range services {
+		if !isContainerRunning(service.Username) {
+			restartServices()
```

Comment:
should this panic if a restartServices() call errors. if so, we should name this function something like `MustEstablishConn`